### PR TITLE
feat: Expose Parquet version and root repetition spec options

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Some old parquet readers like Snowflake imported need the root repetition to be undefined, and they also only support Parquet v1.

This PR exposes both options.

Related to https://github.com/cloudquery/cloudquery-issues/issues/2106 (internal issue)

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: parquet/spec.go
Comment (marianogappa) [inline parquet/spec.go:29]: Should this be a `default`?

**Comment by erezrokah** [comment]
Testing this via https://github.com/cloudquery/cloudquery/pull/18906

**Review by marianogappa — APPROVED** [review]

File: parquet/spec.go
Comment (erezrokah) [reply parquet/spec.go:29]: That's the current default

Comment (marianogappa) [reply parquet/spec.go:29]: I was referring to syntax rather than semantics. The last two return statements are identical.

Comment (marianogappa) [reply parquet/spec.go:29]: If you think it's clearer, that's fair enough.

Comment (erezrokah) [reply parquet/spec.go:29]: > I was referring to syntax rather than semantics. The last two return statements are identical.

Ah got it, yeah I was unsure which way to go. I don't have a strong opinion so I'll update the PR

Comment (erezrokah) [reply parquet/spec.go:29]: Ah OK so the reason I did it like this is otherwise we have:
```go
	switch s.Version {
	case "v1.0":
		return parquet.V1_0
	case "v2.4":
		return parquet.V2_4
	case "v2.6":
		return parquet.V2_6
	default:
		return parquet.V2_LATEST
	}
```

and if we ever change the `default` part it's easy to miss add back `case v2Latest`

Comment (marianogappa) [reply parquet/spec.go:29]: Fair enough

**Comment by erezrokah** [comment]
> Testing this via [cloudquery/cloudquery#18906](https://github.com/cloudquery/cloudquery/pull/18906)

Waiting on a confirmation from a customer before merging this PR

**Pull request merged**
